### PR TITLE
feat: image manifest capability

### LINKED ISSUE #128: Extend the Rust SDK to expose the image manifest capability
policy-evaluator has a new [capability](https://github.com/kubewarden/policy-evaluator/pull/429) which should be exposed by the Rust SDK to allow policies to use it. 

### Acceptance criteria
- SDK must be expose the new capability witch allow user to fetch OCI manifests.

### PR DESCRIPTION
**Pull request by jvanz**
## Description

Adds a new capability function to allow users request the OCI image manifest to the given image name.

Fix #128

### CODE REVIEW

**Comment by jvanz** [comment]
@kubewarden/kubewarden-developers I'm trying to add tests to this new function

**Review by flavio — APPROVED** [review]
LGTM, let's try to create a new policy that makes use of that

**Comment by jvanz** [comment]
Blocked by https://github.com/kubewarden/policy-evaluator/pull/429#issuecomment-1924541219

**Comment by jvanz** [comment]
> LGTM, let's try to create a new policy that makes use of that

https://github.com/kubewarden/policy-server/issues/638#issuecomment-1924482913

File: src/host_capabilities/verification.rs [hunk 1]
```diff
@@ -174,6 +174,7 @@ mod tests {
         // needed for creating mocks
         #[allow(dead_code)]
         pub fn host_call(_binding: &str, _ns: &str, _op: &str, _msg: &[u8]) -> CallResult {
+            println!("verification host_call");
```
Comment (flavio) [inline src/host_capabilities/verification.rs:177]: this has to be dropped

**Review by flavio — CHANGES_REQUESTED** [review]

File: src/host_capabilities/oci.rs [hunk 2]
```diff
@@ -21,3 +37,266 @@ pub fn manifest_digest(image: &str) -> Result<ManifestDigestResponse> {
...
+        match response {
+            OciManifestResponse::Image(image) => {
+                assert_eq!(image.schema_version(), SCHEMA_VERSION);
+                assert_eq!(
+                    image.media_type().clone().expect("missing media type"),
+                    MediaType::ImageManifest
+                );
+                let config = image.config().clone();
+                assert_eq!(config.media_type(), &MediaType::ImageConfig);
+                assert_eq!(config.size(), 7023);
```
[the remark this answers is not in the document]
Comment (jvanz) [reply src/host_capabilities/oci.rs]: Indeed, I've simplified the code by comparing the whole objects.

**Review by viccuad — APPROVED** [review]
LGTM!

File: src/host_capabilities/oci.rs [hunk 3]
```diff
@@ -21,3 +37,189 @@ pub fn manifest_digest(image: &str) -> Result<ManifestDigestResponse> {
...
+                "sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b",
+            ),
+            (
+                73109,
+                "sha256:ec4b8955958665577945c89419d1af06b5f7636b4ac3da7f12184802ad867736",
+            ),
+        ]
+        .iter()
+        .map(|l| {
+            let manifest = DescriptorBuilder::default()
```
Comment (fabriziosestito) [inline src/host_capabilities/oci.rs:155]: nit: this let bining is not necessary, you could return directly

**Review by fabriziosestito — APPROVED** [review]

**Pull request merged**